pkg/awsutils/awsutils.go

```diff
@@ -251,6 +259,24 @@ func (ss *StringSet) Difference(other *StringSet) *StringSet {
 	return &StringSet{data: ss.data.Difference(other.data)}
 }
 
+func (ss *StringSet) Has(item string) bool {
+	ss.RLock()
+	defer ss.RUnlock()
+	return ss.data.Has(item)
+}
+
+// UseCustomNetworkCfg returns whether Pods needs to use pod specific configuration or not.
+func UseCustomNetworkCfg() bool {
+	if strValue := os.Getenv(envCustomNetworkCfg); strValue != "" {
+		parsedValue, err := strconv.ParseBool(strValue)
+		if err == nil {
+			return parsedValue
+		}
+		log.Warnf("Failed to parse %s; using default: false, err: %v", envCustomNetworkCfg, err)
+	}
+	return false
+}
```

Comment:
We shouldn't just copy this from `ipamd.go` and call it every time. It's better to just read it once, then set it as a property. Maybe on the `EC2InstanceMetadataCache` struct, just pass it in `awsutils.New(c.useCustomNetworking)`?

Reply:
Yeah sure makes sense. Since it was read from the environment variable, I implemented a similar function. Will fix it by passing into New as you mentioned.

Reply:
Updated the description with UT.